### DISCUSSION TITLE
How to make a puzzle with individual piece sizes and get it solved

### DISCUSSION THREAD

**Discussion by sluedecke** [flbulgarelli/headbreaker#38] [Q&A] [answered]
I want to create a puzzle for a flag.  This flag has three stripes, all of different widths and I want to have pieces without handles.  Else it would be way to easy to solve.

I use a custom inset:

```javascript
const customNoneInset = {
          isSlot: () => false,
          isTab: () => false,
          isNone: () => true,
          match: () => true,
          toString: () => 'Empty',
          complement: () => customNoneInset,
          serialize: () => 'E',
        };
```

and try to sketch pieces like:

```javascript
this.pp.sketchPiece({
  structure: {
    up: headbreaker.None,
    right: customNoneInset,
    down: headbreaker.None,
    left: customNoneInset,
  },
  metadata: {
    id: 'b',
    targetPosition: { x: 100, y: 0 },
  },
  size: headbreaker.diameter({ x: 200, y: pieceSize.y }),
});
```

(x is the individual piece width, all pieces shall be of`pieceSize.y` height)

This somewhat works, stripes / pieces can be connected, but I don't know which validator to use to have it successfully solved.

Can you help with some example code for this use case?

**Comment by flbulgarelli** [comment, marked as answer]
Hi! I know this may not be best answer - sorry for the delay - , but I'd recommend you to check [this headbreaker-based product](https://mumuki.github.io/mumuki-puzzle-runner/docs/), which offers some additional ideas about how to implement custom validators: https://github.com/mumuki/mumuki-puzzle-runner/blob/master/lib/public/js/muzzle.js 

In particular this piece of code may be of interest to you, since it validates connections manually: 

https://github.com/mumuki/mumuki-puzzle-runner/blob/2554b088c068ae7d0b75dd0dda0867c704b63a80/lib/public/js/muzzle.js#L473-L479
